framework/src/outputs/Output.C

```diff
@@ -117,6 +122,10 @@ Output::Output(const InputParameters & parameters)
     _minimum_time_interval(getParam<Real>("minimum_time_interval")),
     _sync_times(std::set<Real>(getParam<std::vector<Real>>("sync_times").begin(),
                                getParam<std::vector<Real>>("sync_times").end())),
+    _sync_times_object(isParamValid("sync_times_object")
+                           ? dynamic_cast<Times *>(&_problem_ptr->getUserObject<Times>(
```

Comment:
```suggestion
                           ? static_cast<Times *>(&_problem_ptr->getUserObject<Times>(
```
If the casting fails it will fail first in `getUserObject` to it should be safe to `static_cast` here